### DISCUSSION TITLE
Metaprogramming

### DISCUSSION THREAD

**Discussion by RubenVerg** [Vyxal/Vyxal#2066] [Ideas]
Suggestions for stuff related to metaprogramming.

**Comment by RubenVerg** [comment]
### Get/Set Variable from String

Get or set a variable of which the name is specified through a string

**Comment by RubenVerg** [comment]
### Record Fields

All fields of a record, as a list of name/value pairs

Could also do `:ι$Z` for lists

**Comment by RubenVerg** [comment]
### Type Of

The type of a value, as a string description (or maybe returning the constructor for records?)

**Comment by RubenVerg** [comment]
### Constructor

The constructor that was used to create a record

**Comment by ysthakur** [comment]
Maybe a quote structure? And splices within that?

**Comment by RubenVerg** [comment]
### Lambda Properties

Properties of lambdas, ie their arity/name of arguments and their code
